### ISSUE TITLE
X-Session-ID not generated by Squid

### ISSUE THREAD

**Issue by btimby** [nathan-hoad/python-icap#6]
I feel I might be missing something. I cannot find a way to configure Squid to generate and provide an `X-Session-ID` header.

I did find a way to have squid preserve this header (when added by REQMOD) and provide it to RESPMOD (`adaptation_masterx_shared_names`). But this requires the ICAP server to generate and inject the header, Squid simply receives it and returns it with the matching request.

With a number of small changes I was able to get session support working, but if I am missing something and it works without my modifications, I would like to know.

**Comment by nathan-hoad** [comment]
You're not missing something - this was a misunderstanding of Squid's shared name support back when I wrote this. The ICAP server should be generating the session ID header value on response to the REQMOD, which Squid will then put on the RESPMOD request.

**Comment by btimby** [comment]
Good to know. Most of what was needed to get sessions working was already there.

https://github.com/btimby/python-icap/commit/cf82bf7b4af7695d27a9ff1187b8975ffdd0789f

However, I later removed the warning as well since this seemed like the desired behavior.

I have quite a few changes in my fork at this time.

- Case insensitive `del` on HeaderDict.
- Completed session support.
- Added cookie header management (`.cookies`, `.set_cookie()`, `.del_cookie()`. With ability to mask individual cookies.
- Test in virtualenv
- Moved `data/` to `test/data/`. Created missing `.data` file for a failing test.
- Tests pass.

I also did basic linting on the code with flake8.

Are you interested in these changes?

**Comment by btimby** [comment]
I also added `requirements.txt` and `setup.py`. I don't know if you are interested in PyPI, but the `setup.py` helps install with pip.

**Comment by nhoad** [comment]
Yeah sure, may as well send them through. I don't do much work with ICAP anymore, but keeping the library working seems like a reasonable thing to do.

As for pypi, I actually parked the icap name on there years ago, when I had the intent of working on this a lot more.
